pulp_ansible/app/galaxy/v3/filters.py

```diff
@@ -0,0 +1,112 @@
+from django.contrib.postgres.search import SearchQuery
+from django.db.models import fields as db_fields
+from django.db.models.expressions import F, Func
+from django_filters import filters
+from django_filters import FilterSet
+from django.db.models import Q
+
+
+class CollectionVersionSearchFilter(FilterSet):
+    """A custom filterset for cross-repo search."""
+
+    name = filters.CharFilter(method="filter_by_name")
+    namespace = filters.CharFilter(method="filter_by_namespace")
+    version = filters.CharFilter(method="filter_by_version")
+    distribution = filters.CharFilter(method="filter_by_repository_name")
```

Comment:
This should be distribution_base_path. It should look up the distribution, find the associated repository and return it's content.